pyresample/gradient/__init__.py

```diff
@@ -421,7 +421,7 @@ def parallel_gradient_search(data, src_x, src_y, dst_x, dst_y,
                 dst_x[i], dst_y[i],
                 method=method)
             res = da.from_delayed(res, (num_bands, ) + dst_x[i].shape,
-                                  dtype=np.float64)
+                                  dtype=np.float64).astype(arr.dtype)
```

Comment:
Ok, I think we can do better than this. What I see here (and a bit above is that we convert the input data to float64, do the resampling, then convert back to float32.
Using cython fused types, we should be able to template the cython function to accept both float64, float32, or even ints. [This cython doc section](http://docs.cython.org/en/latest/src/userguide/fusedtypes.html#selecting-specializations) should help. I can give you a hand if you don't feel comfortable with this.